content/outcomes/application-development-rhythm/dev-building-blocks-1.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Building Blocks, Part 1
+weight: 30
+layout: single
+team:
+  - VMware Tanzu Labs
+---
+This article is the third in a five-part series.
+It provides a set of building blocks upon which the modern app developer
+can follow a structured and disciplined workflow.
+
+A healthy *Developer Rhythm* uses a disciplined workflow,
+and the workflow requires a set of building blocks to facilitate that
+flow.
+The order reflects the more general building blocks,
+to the more specific.
+
+![Principles, Practices](/images/outcomes/application-development-rhythm/dev-building-blocks.jpg)
+
+*Note:*
+For the remainder of this article,
+the term *developer* either refers to a *single developer*,
+or a *single pair* of developers working on a single workflow.
+
+## What you will learn
+
+By the end of this article you will be able to:
+
+-   List and describe the development *practices* you will use in your
+    day-to-day development flow.
+
+-   List and describe the development flow *techniques and methods* in
+    your day-to-day development flow.
+
+## Coding and Architectural Design Principles and Practices
+
+Building a brand-new application can be relatively easy.
+There is nothing existing to change to accommodate new features,
+and nothing existing to break.
+
+Most of the developer work will be adding and modifying code of
+*an existing codebase*.
+This could imply frequent change.
+Frequent change can be difficult.
+
+Modern app developers must gain knowledge and experience applying
+principles and practices on a continual basis that help solve the
+problem they are trying to solve,
+while also keeping the codebase *flexible to change*.
+
+There are many development principles.
+The following is a good principle that is universal for software,
+regardless of the language or technology.
+
+***Code must be easy for any developer to read, adapt and extend.***
+
+NOTE:
+There are other, specific principles to support this broad one,
+but they are outside the scope of this article.
+
+## Pair Programming Practices
```

Comment:
Are Pairing, TDD, etc sub-topics of "Coding and Architectural Design Principles and Practices"? If so consider making them 3rd-level headings with `### Heading`

Reply:
@joemoore thanks for feedback, fixed